examples/abi/Cargo.toml

```diff
@@ -0,0 +1,24 @@
+[package]
+name = "abi"
+version = "0.1.0"
+authors = ["Near Inc <hello@nearprotocol.com>"]
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+near-sdk = { path = "../../near-sdk", features = ["unstable"] }
+serde = { version = "1", features = ["derive"] }
+schemars = "0.8"
```

Comment:
So maybe I am missing something but this is conflicting with your other comment about removing re-export of schemars: we need to derive `schemars::JsonSchema` for all types that are going to show up in ABI. I can create an abstraction for developers and make them use a macro to derive *something* that gives them the ability to generate ABI for these types:

```rust
#[proc_macro_attribute]
pub fn derive_near_abi(_attr: TokenStream, input: TokenStream) -> TokenStream {
    TokenStream::from(quote! {
        #[derive(schemars::JsonSchema)]
        #input
    })
}
```

The problem here is that developers will still need to have `schemars` as a dependency, right? I could use `near_sdk::schemars` here instead but you are saying we should remove it in the other thread. I also explored the option of creating my own trait `NearAbi` that would encapsulate JSON Schema logic, but unfortunately not having `schemars::JsonSchema` directly means I cannot use `schemars::Generator` which is a major problem when it comes to ABI generation.

Reply:
The idea I'm trying to suggest is having some wrapper type which internally uses schemars, so that if we change the internal functionality it doesn't change the API for developers

Reply:
Discussed with @austinabell offline and he is going to explore this separately